### ISSUE TITLE
Add a Symfony Console Command to run Commands

### ISSUE THREAD

**Issue by matks** [PrestaShop/PrestaShop#24571]
#### Describe the solution you'd like

Add a Symfony Command to run Commands from the Console

Example
```
$ bin/console prestashop:command <command> <...parameters>

$ bin/console prestashop:command SwitchDebugModeCommand true
```

**Comment by jolelievre** [comment]
So the question is how do we format the parameters in input? Should it a json format?
Or just a list of inline parameters (maybe inside "" when you need to insert spaces)?

Also I guess we'll need a mini component which is able to parse the input parameters and especially cast them into the proper type. Easy to do when the command is fully typed, but we'll have to handle loose types defined in PHPDoc as well).

And I hope we only use built-in types as inputs in our commands and not Objects, or the parsing will become more complex

**Comment by jolelievre** [comment]
Another detail, is the class name enough? I'm thinking about namespaces, especially in case we would have commands with same names but different namespaces?

**Comment by matks** [comment]
> Another detail, is the class name enough? I'm thinking about namespaces, especially in case we would have commands with same names but different namespaces?

I guess that the must user-friendly solution would be:

- user can use a classname when it is unique, the Command is able to find the right Command php class
- if user inserts a classname that is not unique, the Command returns an error message and ask for a FQDN
- using a FQDN command name also works if classname is unique 😄

**Comment by jolelievre** [comment]
Yes an error message displaying the list of conflicting class would be nice

**Comment by matks** [comment]
Done in 1.7.5
